themes/classic/templates/customer/_partials/order-detail-no-return.tpl

```diff
@@ -42,36 +42,38 @@
 
   <tfoot>
     {if $priceDisplay && $use_tax}
-      <tr>
+    <tr>
         <td>{l s='Items (tax excl.)'}</td>
-        <td colspan="4" class="text-xs-right">{$order.data.total_products}</td>
-      </tr>
+        <td colspan="4" class="text-xs-right">{$order.amounts.total.amount}</td>
+    </tr>
     {/if}
+    {if $order.amounts.subtotals.tax}
```

Comment:
@maximebiloe you must have to check for undefined value then, but you don't know why the value is undefined.